relayer/message_relayer.go

```diff
@@ -247,13 +247,24 @@ func (r *messageRelayer) createSignedMessageAppRequest(requestID uint32) (*avala
 	for node := range nodeValidatorIndexMap {
 		nodeIDs.Add(node)
 	}
+	connectedNodes := r.relayer.network.ConnectPeers(nodeIDs)
 
-	// TODO: We may still be able to proceed with signature aggregation even if we fail to connect to some peers.
-	// 		 We should check if the connected set represents sufficient stake, and continue if so.
-	_, err = r.relayer.network.ConnectPeers(nodeIDs)
-	if err != nil {
+	// Check if we've connected to a stake threshold of nodes
+	connectedWeight := uint64(0)
+	for node := range connectedNodes {
+		connectedWeight += validatorSet[nodeValidatorIndexMap[node]].Weight
+	}
+	if !utils.CheckStakeWeightExceedsThreshold(
+		big.NewInt(0).SetUint64(connectedWeight),
+		totalValidatorWeight,
+		r.warpQuorum.QuorumNumerator,
+		r.warpQuorum.QuorumDenominator,
+	) {
 		r.relayer.logger.Error(
-			"Failed to connect to peers",
+			"Failed to connect to a threshold of stake",
+			zap.Uint64("connectedWeight", connectedWeight),
+			zap.Uint64("totalValidatorWeight", totalValidatorWeight),
+			zap.Any("warpQuorum", r.warpQuorum),
```

Comment:
`err` referenced on lines 268 and 270 is no longer referencing a relevant error.

Reply:
Fixed